src/foam/core/AbstractStringPropertyInfo.java

```diff
@@ -94,4 +94,11 @@ public boolean isDefaultValue(Object o) {
   public void format(foam.lib.formatter.FObjectFormatter formatter, foam.core.FObject obj) {
     formatter.output(get_(obj));
   }
+
+  public Object castObject(Object value) {
+    if ( value instanceof Number ) {
```

Comment:
Couldn't you just always return value.toString() (if it isn't null)? Since toString() on a String will just return itself. And then other types will work also.